### PR TITLE
Add migration notice and translation banner to the blank layout; fix translation banner logic

### LINKED ISSUE #784: Blank layout doesn't show the translation banner or CC referral banner
## Description

The blank layout, used on the home and 404 pages, doesn't show the incomplete translation banner _or_ the cc referral banner.

## Reproduction

1. Visit the homepage on a not fully translated locale and/or with ?referrer=creativecommons.org in the url
2. Observe the banners are not present

## Environment

 - Device: 
 - OS: 
 - Browser: 
 - Version: 
 - Other info: 

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #784 by @zackkrida

## Description

This PR:
- adds the migration notice and translation banner to the blank layout (used on the homepage and 404)
- changes the logic to detect whether we should show the translation banner. Since we moved the locale detection to path-based, we remove the WordPress theme - Nuxt locale passing messages, and use the app i18n locale to detect whether we should show the banner.
- adds the simple e2e tests for the translation banner.

## Testing Instructions

Run the app and go to `http://localhost:8443/ru/`. You should be able to see the translation banner at the top. From there, you can go to the Russian Openverse translation page, or close the banner. If you navigate to other pages, you should not see the banner after closing it once. Also, you shouldn't see the banner at `http://localhost:8443`.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: test/e2e/translation-banner.spec.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+test('Can close the translation banner', async ({ page }) => {
+  await page.goto('/ru/search/')
+  await page.click('[aria-label="Закрыть"]')
+
+  const banner = page.locator(
+    'text=The translation for Russian locale is incomplete. Help us get to 100 percent by '
+  )
+  await expect(banner).not.toBeVisible({ timeout: 10 })
+})
```
Comment (sarayourfriend) [inline test/e2e/translation-banner.spec.js:28]: Maybe one more test to verify that the banner is automatically hidden on subsequent page loads after dismissal?

**Review by sarayourfriend — APPROVED** [review]
LGTM! Works well. Just a couple non-blocking comments. Glad we can fix this :tada: 

Should we also create a follow-up issue to resolve once this and the theme updates are all deployed to remove any vestiges of the cookie based local stuff? For example I think `en_US` might be redundant with `en` in that check as we're never given the `wpLocale`, just the slug from the theme.

**Review by zackkrida — APPROVED** [review]
LGTM. On the home and 404 pages these elements cause the height of the page to exceed 100vh, but I think that's okay. 

I'll make separate issues, but both banners enabled is not particularly attractive (cc @panchovm):

The translation banner styles aren't ideal on mobile, I think it can be made smaller with some adjustment.

CleanShot 2022-02-18 at 12 25 10@2x

Both banners enabled looks pretty ugly, honestly:

CleanShot 2022-02-18 at 12 26 27@2x

File: src/composables/use-i18n-sync.js [hunk 2]
```diff
@@ -1,62 +1,34 @@
...
-    // Always set `dir` property, default to 'ltr'
-    document.documentElement.dir = locale.dir
-  }
   /**
    * Show banner inviting to contribute translations if fewer than 90%
    * of strings are translated for current locale.
    * Hard-coded to false for default locales: `en` and `en_US`.
    *
-   * @param {import('../locales/scripts/types').i18nLocaleProps} locale - current locale object
+   * @param {string} localeCode - the wpLocale code for the current locale
```
[the remark this answers is not in the document]
Comment (obulat) [reply src/composables/use-i18n-sync.js:23]: This is actually not a nit, as there are at least three pieces to each locale that are important here, and we should be sure to select the correct one.
In the `wporg_locales.sql` you can see that each locale has `locale` and `subdomain` properties. `locale` corresponds to the `wp_locale` property in the `valid-locales.json` file. This is different from the key (`code`) that we use for the locale in Vue-i18n (`valid-locales.json` file), and is the same as the `slug` in the GlotPress list of locales where we originally take the data from. And that `slug`, in turn, is different from the `subdomain` that is used on WordPress.org. 
This is really complicated, and all the values are just a little bit different from one another for some locales.

So, what does `locale_slug` mean in this case, @sarayourfriend ? Is it the same as the `subdomain` in the `wporg_locales.sql`, or `slug` from the (GlotPress list)[https://github.com/GlotPress/GlotPress-WP/blob/develop/locales/locales.php]?
You can see the difference if you check Venezuelan Spanish: it's wporg subdomain is `ve`, slug in GllotPress is `es-ve`, wpLocale is `es_VE`, and the HTML lang code for it should be `es-ve` (although ve.wordpress.org site uses `es`)

[the remark this answers is not in the document]
Comment (sarayourfriend) [reply src/composables/use-i18n-sync.js:23]: So if you look at the `GP_Locales` class in the `locales` plugin for wp.org it's a little easier to see what I mean. `wporg_locales.sql` I think just configures the subdomain -> wp_locale mapping.

https://github.com/wordpress/wordpress.org/blob/e38ba4261c11f90443116df4b2e0654410eb8c04/wordpress.org/public_html/wp-content/mu-plugins/pub/locales/locales.php#L929:L938

There you can see that the string _we_ use to match a `<locale>.json` is the `slug` property. `get_locale()` in WordPress returns the `wp_locale` property.

The subdomain is totally separate. The Brazilian locale, for example, is subdomained at `br.wordpress.org`, but the locale is `pt_BR` and the slug is `pt-br`. The subdomains that wp.org uses are really just kind of made up/convenient compared to the proper locale codes they map to. For example, the Brazilian `br` subdomain actually clashes with the Breton locale code: https://github.com/wordpress/wordpress.org/blob/e38ba4261c11f90443116df4b2e0654410eb8c04/wordpress.org/public_html/wp-content/mu-plugins/pub/locales/locales.php#L410

Weird stuff! :shrug:

[the remark this answers is not in the document]
Comment (obulat) [reply src/composables/use-i18n-sync.js:23]: Thank you for your detailed explanation!
We also use this `slug` property to create the translate link, as in `(https://translate.wordpress.org/locale/{slug}/default/meta/openverse/)`. When first writing the translation banner logic, I was afraid we would need the subdomain data as well somewhere. But if we only use slug, we can simplify the locale data object that we have :)

Comment (obulat) [inline src/composables/use-i18n-sync.js:23]:
```suggestion
   * @param {string} localeCode - the slug for the current locale
```

**Comment by panchovm** [comment]
I will work on both banners to match the WP.org style. Thanks @obulat for the work and @zackkrida for tagging me

**Pull request merged**